### DISCUSSION TITLE
Google Play Subscriptions are always in "Payment Pending" when free-trial expires.

### DISCUSSION THREAD

**Discussion by uncleandyhere** [hyochan/react-native-iap#1413] [Q&A]
> Please use [Discussion board](https://github.com/dooboolab/react-native-iap/discussions) if you want get some help out of it. Please report issue if there is a bug.

### Version of react-native-iap
5.2.14

### Version of react-native
0.63.4

### Platforms you faced the error (IOS or Android or both?)
Android (no IOS app)

### Expected behavior
Successfully charges when free-trial expires

### Actual behavior
After 3-day free-trial, the order status is always "Payment pending".
After one month, the "Payment pending" status becomes  "Payment declined"

### Tested environment (Emulator? Real Device?)
Real Device, Real Users

### Steps to reproduce the behavior
Buy the subscription with free-trial.

---

I followed the implement guide. 
My product is a subscription with 3-day free trial. (Grace Period: None) 
And I haved test the flow with tester accounts using google's fake card, it is successful.
<img width="1195" alt="螢幕擷取畫面 2021-07-11 174829" src="https://user-images.githubusercontent.com/87260123/125190546-4cf0dc00-e270-11eb-8f1d-e35a5a4253ac.png">

But if the real cusomters buy the subscription with free-trial, the order is always "Payment pending" after 3-day free-trial expires.

<img width="1194" alt="螢幕擷取畫面 2021-07-11 161825" src="https://user-images.githubusercontent.com/87260123/125190689-f932c280-e270-11eb-9dd9-b089f2c8a063.png">

The "Payment pending" status lasts for 1 month, then it becomes "Payment declined".

<img width="1208" alt="螢幕擷取畫面 2021-07-11 175115" src="https://user-images.githubusercontent.com/87260123/125190623-a822ce80-e270-11eb-8a8a-7f95bcdcff52.png">

But if I provide subscription item WITHOUT free-trial, it can be charged successfully and immediately.
<img width="1235" alt="螢幕擷取畫面 2021-07-11 175727" src="https://user-images.githubusercontent.com/87260123/125190810-968df680-e271-11eb-8f08-73873c4d6c01.png">

All of my ~30 orders (subscription with free-trial) fails to charge at auto-renewing when free-trial expires.
These orders are from real customers from more than 4 different countries.

Here is my main code:   
```
export const IAP_init = ()=>{
    RNIap.initConnection().then(() => {
        
        RNIap.flushFailedPurchasesCachedAsPendingAndroid().catch(() => {
            
        }).then(async () => {
            RNIap.purchaseUpdatedListener(My_purchaseUpdatedListener);
            RNIap.purchaseErrorListener(My_purchaseErrorListener);
         }
}
```

```
const My_purchaseUpdatedListener = async (purchase) => {
    const receipt = purchase.transactionReceipt;
    if (receipt) {
                let finish_result = await RNIap.finishTransaction(purchase, false);
    }
}
```

I don't implement any server side receipt-validation facilities for simplicity.

---

I find a similar issue in stackoverflow, but there is no any solid answer so far.
[https://stackoverflow.com/questions/60994807/why-there-are-so-many-pending-payments-for-subscription-from-google-play-console](https://stackoverflow.com/questions/60994807/why-there-are-so-many-pending-payments-for-subscription-from-google-play-console)

I don't know what's wrong with my integration.
And I really need the help.
Thank you

**Comment by andresesfm** [comment]
Are you calling `acknowledge`?

This is from the Android documentation

> Note: If you do not acknowledge a purchase within three days, the user automatically receives a refund, and Google Play revokes the purchase.

**Reply by uncleandyhere** [reply]
@andresesfm Tnanks for your reply.
Yes I am, and I also checked the purchase info. The `isAcknowledgedAndroid` attribute is `true`, and `purchaseStateAndroid` is `1`.

BTW, I notice that all these failed orders are from countries which have "google gift card" as one of the paying methods.
Maybe all of my customers buying the free-trial subscription using "google gift card" but never go to convenience store?
Because I have one order from a country which didn't support gift card, and the order is charged successfully !
So my next step might be cancelling all free-trial in those countries which support gift card.

Any other suggestion will be welcome. Thanks a lot!

**Reply by brunogimenes** [reply]
Were you able to solve this issue?

I am having the exact same issue. Around 60, 70 transactions per week, all of them gets in a "pending state" after trial ends and after some days it gets denied.

It looks very improbable that all of these users have insufficient funds or are using gift cards as payment methods.

Thaanks!

**Reply by tilpap** [reply]
Did you find any solution to this? Same problem for me
